include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -38,6 +38,11 @@ namespace emulated
             output.y = uint32_t(product & 0x00000000FFFFFFFFull);
             return output;
         }
+        
+        bool isNaN64(uint64_t val)
+        {
+            return bool((0x7FF0000000000000ull & val) && (0x000FFFFFFFFFFFFFull & val));
+        }

```

Comment:
start a header called `tgmath.hlsl` and add a `nbl::hlsl::isnan` there (also remove `core::isnan` from the C++ header at the same time)